scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,546 @@
+    val pomFile = URL(pomFileUrl).openStream().bufferedReader().readText()
+    val pomText = pomFile
+    var cursor = -1
+    if (pomText.length > 11) {
+      for (index in 0..(pomText.length - 11)) {
+        if (pomText.substring(index, index + 10) == LICENSES_TAG) {
+          cursor = index + 9
+          break
+        }
+      }
+      if (cursor != -1) {
+        var cursor2 = cursor
+        while (cursor2 < (pomText.length - 12)) {
+          if (pomText.substring(cursor2, cursor2 + 9) == LICENSE_TAG) {
+            cursor2 += 9
+            while (cursor2 < pomText.length - 6 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 6
+              ) != NAME_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 6
+            val url = StringBuilder()
+            val urlName = StringBuilder()
+            while (pomText[cursor2] != '<') {
+              urlName.append(pomText[cursor2])
+              ++cursor2
+            }
+            while (cursor2 < pomText.length - 4 &&
+              pomText.substring(
+                cursor2,
+                cursor2 + 5
+              ) != URL_TAG
+            ) {
+              ++cursor2
+            }
+            cursor2 += 5
+            while (pomText[cursor2] != '<') {
+              url.append(pomText[cursor2])
+              ++cursor2
+            }
+            licenseNamesFromPom.add(urlName.toString())
+            licenseLinksFromPom.add(url.toString())
+            licenseList.add(
+              License
+                .newBuilder()
+                .setLicenseName(urlName.toString())
+                .setPrimaryLink(url.toString())
+                .setPrimaryLinkType(PrimaryLinkType.PRIMARY_LINK_TYPE_UNSPECIFIED)
+                .build()
+            )
+          } else if (pomText.substring(cursor2, cursor2 + 12) == LICENSES_CLOSE_TAG) {
+            break
+          }
+          ++cursor2
+        }
+      }
+    }
+    val mavenDependency = MavenDependency
+      .newBuilder()
+      .setArtifactName(it.coord)
+      .setArtifactVersion(artifactVersion.toString())
+      .addAllLicense(licenseList)
+      .setOriginOfLicense(OriginOfLicenses.UNKNOWN)
+
+    mavenDependencyList.add(mavenDependency.build())
+  }
+  return MavenDependencyList.newBuilder().addAllMavenDependency(mavenDependencyList).build()
+}
+
+fun writeTextProto(
+  pathToTextProto: String,
+  mavenDependencyList: MavenDependencyList
+) {
+  val file = File(pathToTextProto)
+  val list = mavenDependencyList.toString()
+
+  file.printWriter().use { out ->
+    out.println(list)
+  }
+}
+
+fun runMavenRePinCommand(rootPath: String) {
+  val rootDirectory = File(rootPath).absoluteFile
+  val bazelClient = BazelClient(rootDirectory)
+  val output = bazelClient.executeBazelRePinCommand(
+    "bazel",
+    "run",
+    "@unpinned_maven//:pin"
+  )
+  println(output)
+}
+
+private class BazelClient(private val rootDirectory: File) {
```

Comment:
Prefer to use the BazelClient in scripts/.../common, instead.

Reply:
Tried using the new BazelClient, but it is giving an error - 
```
Exception in thread "main" java.lang.IllegalStateException: Expected non-zero exit code (not 7) for command: [bazel, query, 'deps(deps(//:oppia), intersect, //third_party/...), intersect, @maven//...'].
Standard output:

Error output:
Loading: 0 packages loaded
ERROR: The package part of 'deps(deps(//:oppia) intersect //third_party/...) intersect @maven//...' should not end in a slash
Loading: 0 packages loaded
Loading: 0 packages loaded
        at org.oppia.android.scripts.common.BazelClient.executeBazelCommand(BazelClient.kt:146)
        at org.oppia.android.scripts.common.BazelClient.executeBazelCommand$default(BazelClient.kt:135)
        at org.oppia.android.scripts.common.BazelClient.retrieveThirdPartyMavenDependenciesList(BazelClient.kt:122)
        at org.oppia.android.scripts.maven.GenerateMavenDependenciesListKt.runBazelQueryCommand(GenerateMavenDependenciesList.kt:275)
        at org.oppia.android.scripts.maven.GenerateMavenDependenciesListKt.main(GenerateMavenDependenciesList.kt:50)

```

I also encountered this before, so I made some changes in `executeCommand()` method of the private Bazel Client, and then it worked.

Original `executeCommand()` method - 
```kotlin
val process =
      ProcessBuilder(assembledCommand)
        .directory(workingDir)
        .redirectErrorStream(includeErrorOutput)
        .start()
```

After changes in the private BazelClient -
```kotlin
val process = ProcessBuilder()
      .command("bash", "-c", command)
      .directory(workingDir)
      .redirectErrorStream(includeErrorOutput)
      .start()
```

Reply:
Fixed.

Reply:
Verified.